General alcohol tweaks

TheVekter opened the pull request:
- Tweaks shot glasses. They can hold 15 units now, so certain shots can be mixed in them without having to use a shaker. Player now drinks all 15 units at once.
- Adds special sprites for rum, vodka, and whiskey (3 most likely boozes for shots). This lets us port certain shot-based drinks (B-52 comes to mind) to shot glass size.
- Ported B-52 and Toxins Special to shot glasses. You can no longer mix them in normal glasses.
- Adds a generic iconstate for everything else.
- Fixes #7625 
- Fixes an unreported bug - apparently I broke the tequila bottle sprite in #7568.

NOTE: The drunkenness tweaks have been removed, as part of #7439 takes care of this issue.

TheVekter committed:
Fixes shot glass sprite changing when something's poured in (TEMP FIX)

TheVekter committed:
Adds sprites and iconstates for shot glass.

TheVekter committed:
Adds multiple drinks to the shotglass list so they can be mixed within.

Re-adds bottle of booze that got removed in a merge conflict a while back.

TheVekter committed:
Wow apparently I broke the tequila sprite

Tenebrosity commented:
So this uses different sprites to represent different drinks in the same glass instead of changing the colour of the reagent?

TheVekter commented:
Yeah. This is pretty similar to how the drinking glass is coded. It's
because there are specific sprites for certain drinks, as opposed to
beakers where it just shows the color of the reagent.

If you pour something like plasma or choral into a drinking glass it shows
up as a glass full of brown liquid and calls it a "glass of what".
On Feb 10, 2015 1:46 AM, "homsar" <notifications@github.com> wrote:

> So this uses different sprites to represent different drinks in the same
> glass instead of changing the colour of the reagent?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/7635#issuecomment-73657304>
> .
>

RemieRichards commented on code/modules/food&drinks/drinks/drinks/drinkingglass.dm:
```diff
@@ -162,6 +162,7 @@
 				icon_state = "tequilasunriseglass"
 				name = "tequila Sunrise"
 				desc = "Oh great, now you feel nostalgic about sunrises back on Terra..."
+			// Toxins Special is also in drinks.dm because shot glasses, mirror any changes there. //
```
the first sign that this is not the right way to go about this.

TheVekter replied:
I'm open to suggestions. I honestly considered removing Toxins Special from glasses entirely, but left it in as a quality of life thing (to not confuse people when their drinks suddenly wouldn't mix).

RemieRichards replied:
it's less specifically the drinks themselves and the fact that you're having to ask for future changes to be mirrored, hint, nobody is going to do so and if they do then they are a really nice person.

These drinks appear with shotglass icons, now we have actual shotglasses you should just remove the ability to mix said drinks in normal glasses.

TheVekter replied:
I'll need to make a changelog for it, but alright. I'll keep an eye out and see if anyone adminhelps about their drinks not working.

RemieRichards replied:
I believe changelogs are becoming more of a mandatory thing so you'll probably need one anyways.
And as with changes it'll probably take a little bit of time for people to learn, a changelog entry would help this.

TheVekter committed:
Made Toxins Special and B-52 exclusive to shot glasses.
Tweaked drunkenness to make it less effective at lower booze levels, but moreso at higher.
Changelog.

TheVekter committed:
Moved shotglasses back to drinkingglass.dm. Again.
Added a bunch more iconstates for each drink in the booze-o-mat. Any drink the shot doesn't recognize comes up as a "shot of... what?"

TheVekter committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into shotglasseshowdotheywork

TheVekter commented:
I wanna do a bit more testing on this before we merge, but I'm pretty sure this is g2g.

TheVekter commented:
Alright, looks like everything's working on my end. One last commit and we'll be good.

TheVekter committed:
Reverts drunkenness changes per #7439
Fixes a typo
Makes players drink full shot no matter how much is in glass

paprka commented:
the map edits are fucked in this, you removed the janicart

TheVekter committed:
Mapping is hard

TheVekter committed:
Revert "Mapping is hard"
